python/mlx/nn/layers/quantized.py

```diff
@@ -268,3 +268,93 @@ def from_linear(
             ql.bias = linear_layer.bias
 
         return ql
+
+
+class QQLinear(Module):
+    """Quantizes the input and applies an affine transformation using quantized weights.
+
+    Two use cases are supported:
+
+    1) **Inference / frozen weights**: weights are stored in quantized form together with
+       their scales (``self.weight`` is quantized and ``self.scales`` is provided).
+
+    2) **Training / weights are included in gradient computation**:
+        weights are stored in higher precision and are quantized on
+        the fly during computation.
+
+    Compared to the :class:`mlx.nn.QuantizedLinear` layer, this layer
+    quantizes the input as well and includes weights in gradient computations.
+
+    :obj:`QQLinear` also provides the class method :meth:`from_linear` to convert
+    :class:`mlx.nn.Linear` layers to :obj:`QQLinear` layers.
+
+    Note: This layer does not support a bias term yet.
+
+    Args:
+        input_dims (int): The dimensionality of the input features.
+        output_dims (int): The dimensionality of the output features.
+        group_size (int, optional): The group size to use for the quantized weight.
+            See :func:`~mlx.core.quantize`. Default: ``16``.
+        bits (int, optional): The bit width to use for the quantized weight.
+            See :func:`~mlx.core.quantize`. Default: ``4``.
```

Comment:
I think it makes sense to default these to `None` so that the right default for the given `mode` is chosen.

Reply:
(I realize that it's like this for the other Quantize modules and I think I just forgot to change those when I changed the op defaults, but we should change them as well)